src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const strSeparator = typeof separator === 'string'
+      ? separator
+      : separator + '';
```

Comment:
It's 100% correct, but we can even make it shorter by always turning into a string, regardless of its type)
```suggestion
    const strSeparator = String(separator);
```

Reply:
It won't break a program, only make it shine even more😎